### ISSUE TITLE
loss function

### ISSUE THREAD

**Issue by DdChew** [dylan-campbell/bpnpnet#5]
Hi,dylan ！
Train this network with my own data set，in the first few epochs（Lp is not introduced ),Lc fluctuates around 0.8, no obvious drop（Maybe this is normal?）.
I saw that in your paper, the Lc training was conducted for 120 epochs.
Could you tell me the approximate value of Lc after you complete these 120 periods of training，this can give me a reference，thanks！！

**Comment by SergioRAgostinho** [comment]
Your experience is consistent with mine while training on megadepth. The (train and validation) correspondence probability value after 120 epochs for that one is around 0.76. This is sufficient to recreate the results published on the paper.

**Comment by dylan-campbell** [comment]
Hi @DdChew, there should be a noticeable drop in the first few epochs. For megadepth, Lc should be dropping down from about 0.9 to 0.6 after 120 epochs on the validation set (or 0.99 to 0.5 on the training set). If it's fluctuating on your dataset, it's probably not learning well - perhaps try a larger learning rate. We actually found that we get better performance by increasing the learning rate from what we used in the paper, up to 1e-3 for MegaDepth.

@SergioRAgostinho, was this on the same (random) subset of megadepth that we used or a different one? There may be some variation there.

**Comment by dylan-campbell** [comment]
@DdChew: also, if your dataset has a high proportion of outliers it's likely to be harder to learn the inlier correspondences. In this case it'd be helpful to filter some of the outliers first.

**Comment by DdChew** [comment]
@dylan-campbell ,thank you for sharing the data.
I will adjust hyperparameters like learning rate ,try to achieve this result.

**Comment by SergioRAgostinho** [comment]
> @SergioRAgostinho, was this on the same (random) subset of megadepth that we used or a different one? There may be some variation there.

I was using the supplied preprocessing data. I  just ran a training keeping everything default.

**Comment by DdChew** [comment]
Hi ！@dylan-campbell ，at line 117 of posses.py，maybe this correspondenceMatrix should be correspondenceMatrices?

**Comment by dylan-campbell** [comment]
Thanks @DdChew, well-spotted! Fixed.
